pkg/configmap/wsvalidator/wsvalidator.go

```diff
@@ -92,6 +94,93 @@ func GenerateWindowsServiceConfigMap(name, namespace string) *core.ConfigMap {
 	return cm
 }
 
+// ParseAndValidate converts ConfigMap data into the objects representing a Windows services ConfigMap and
+// ensures the data adheres to the expected schema
+func ParseAndValidate(data map[string]string) (*[]FileInfo, *[]Service, error) {
```

Comment:
Consider splitting this behavior in two functions:

- `func GetServices(data map[string]string) (*[]Service, error) {...}`
- `func GetFiles(data map[string]string) (*[]FileInfo, error) {...}`

Reply:
I was thinking that returning the Service and FileInfo slices will be useful in the future, but I can now see the angle that this function is doing too much. 
I may split this into 3 functions: the two Jose mentioned so someone can get each component if needed, and then a top level ParseAndValidate which uses getServices() and getFiles() as helpers

Reply:
I prefer ParseAndValidate function just returns bool. Other functionalities can be split into different functions.